homeassistant/components/websocket_api/http.py

```diff
@@ -89,33 +90,40 @@ def description(self) -> str:
     async def _writer(self) -> None:
         """Write outgoing messages."""
         # Exceptions if Socket disconnected or cancelled by connection handler
-        to_write = self._to_write
+        message_queue = self._message_queue
+        queue_ready = self._queue_ready
         logger = self._logger
         wsock = self.wsock
         try:
             with suppress(RuntimeError, ConnectionResetError, *CANCELLATION_ERRORS):
                 while not self.wsock.closed:
-                    if (process := await to_write.get()) is None:
+                    await queue_ready.wait()
```

Comment:
We could use a future here instead and avoid the whole asyncio event.  That would make this even leaner